pkg/registry/service/ipallocator/allocator.go

```diff
@@ -189,31 +163,32 @@ func (r *Range) Has(ip net.IP) bool {
 		return false
 	}
 
-	r.lock.Lock()
-	defer r.lock.Unlock()
-
-	return r.allocated.Bit(offset) == 1
+	return r.alloc.Has(offset)
 }
 
 // Snapshot saves the current state of the pool.
-func (r *Range) Snapshot() (*net.IPNet, []byte) {
-	r.lock.Lock()
-	defer r.lock.Unlock()
-
-	return r.net, r.allocated.Bytes()
+func (r *Range) Snapshot(dst *api.RangeAllocation) error {
+	snapshottable, ok := r.alloc.(allocator.Snapshottable)
+	if !ok {
+		return fmt.Errorf("Not a snapshottable allocator")
```

Comment:
Error strings should start with a lowercase character (so they can be fragmentized) - couple of other uses in this code